### PR TITLE
:seedling: lint update

### PR DESCRIPTION
**Pull request by perdasilva**
# Description

Updates golangci-lint to its latest version and updates the config to remove a todo.
Adds a Makefile target `make fix-fmt` that runs golangci-lint --fix to (hopefully) fix (most) of the issues automagically.
Addresses a gosec  lint issue where the perms for a file (in a directory that the operator-controller creates) were too broad (0777) -> 0600. I think this should be ok.

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -109,6 +109,10 @@ verify: tidy fmt vet generate manifests #HELP Verify all generated code is up-to
 fmt: #EXHELP Run go fmt against code.
 	go fmt ./...
 
+.PHONY: fix-fmt
+fix-fmt: $(GOLANGCI_LINT) #EXHELP Runs golangci-lint with --fix flag to fix linting issues.
+	$(GOLANGCI_LINT) run --fix $(GOLANGCI_LINT_ARGS)
+
```
Comment (tmshort) [inline Makefile:115]: This sounds an awful lot like the `fmt` target.

Comment (perdasilva) [reply Makefile:115]: Is that a comment or a suggestion?

Comment (perdasilva) [reply Makefile]: merged the targets - is that what you wanted?

**Review by tmshort — APPROVED** [review]
/lgtm

**Pull request merged**